spec/models/labware_creators/donor_pooling_plate_spec.rb

```diff
@@ -621,44 +674,42 @@
   end
 
   describe '#tag_depth_hash' do
+    let(:number_of_pools) { 2 }
     it 'returns a hash mapping positions of wells in their pools' do
-      well_p1_w1 = well = parent_1_plate.wells[0]
-      well.state = 'passed'
-      well.aliquots.first.study = study_1
-      well.aliquots.first.project = project_1
-      well.aliquots.first.sample.sample_metadata.donor_id = 1
-
-      well_p1_w2 = well = parent_1_plate.wells[1]
-      well.state = 'passed'
-      well.aliquots.first.study = study_1
-      well.aliquots.first.project = project_1
-      well.aliquots.first.sample.sample_metadata.donor_id = 2
+      wells = Array(parent_1_plate.wells[0..4]) + Array(parent_2_plate.wells[0..4])
+      wells.each_with_index do |well, index|
+        well.state = 'passed'
+        well.aliquots.first.study = study_1 # same study
+        well.aliquots.first.project = project_1 # same project
+        well.aliquots.first.sample.sample_metadata.donor_id = index + 1 # different donors
+      end
 
-      well_p2_w1 = well = parent_2_plate.wells[0]
-      well.state = 'passed'
-      well.aliquots.first.study = study_1
-      well.aliquots.first.project = project_1
-      well.aliquots.first.sample.sample_metadata.donor_id = 3
-
-      well_p2_w2 = well = parent_2_plate.wells[1]
-      well.state = 'passed'
-      well.aliquots.first.study = study_1
-      well.aliquots.first.project = project_1
-      well.aliquots.first.sample.sample_metadata.donor_id = 1 # same donor as well_p1_w1
+      wells[1].state = 'passed'
+      wells[1].aliquots.first.study = study_1
+      wells[1].aliquots.first.project = project_1
+      wells[1].aliquots.first.sample.sample_metadata.donor_id = 1 # same donor as the first well
 
       subject.build_pools
-      expect(subject.tag_depth_hash[well_p1_w1]).to eq('1')
-      expect(subject.tag_depth_hash[well_p1_w2]).to eq('2')
-      expect(subject.tag_depth_hash[well_p2_w1]).to eq('3')
-      expect(subject.tag_depth_hash[well_p2_w2]).to eq('1')
+      expect(subject.tag_depth_hash[wells[0]]).to eq('1') # 10-well-A1
+      expect(subject.tag_depth_hash[wells[1]]).to eq('1') # 10-well-B1
```

Comment:
Tag depth must be increasing sequentially within the same pool; simply it is showing the position of the well in the pool. It looks like a tag clash here.